runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -220,15 +220,18 @@ public DirectPipelineResult run(Pipeline pipeline) {
    */
   @SuppressWarnings("rawtypes")
   private List<PTransformOverride> defaultTransformOverrides() {
-    return ImmutableList.<PTransformOverride>builder()
-        .add(
-            PTransformOverride.of(
-                PTransformMatchers.writeWithRunnerDeterminedSharding(),
-                new WriteWithShardingFactory())) /* Uses a view internally. */
-        .add(
-            PTransformOverride.of(
-                PTransformMatchers.classEqualTo(CreatePCollectionView.class),
-                new ViewOverrideFactory())) /* Uses pardos and GBKs */
+    ImmutableList.Builder<PTransformOverride> builder =
+     ImmutableList.<PTransformOverride>builder();
+    if (!options.isUnitTest()) {
```

Comment:
Specifically, tests executions for the sdk and runners-core modules (e.g. the `validates-runner` execution) don't need to modify any behavior of the original transforms, and I'd like to avoid having the transform specify how it is permitted to be overridden.

I would like to pull any test-related options out of `DirectOptions` and mark them as `@Internal` and `@Hidden`, however.